README_migrationGuide.md

```diff
@@ -0,0 +1,59 @@
+A short guide to migrating to go-ipld-prime from other repos
+============================================================
+
+Here are some quick notes on APIs that you might've been using
+if you worked with IPLD in golang before go-ipld-prime,
+and a pointer to what you should check out for an equivalent
+in order to upgrade your code to use go-ipld-prime.
+
+(Let us know if there are more pointers we should add to this list to ease your journey,
+or someone else's future journey!)
+
+- Were you using [ipfs/go-datastore](https://pkg.go.dev/github.com/ipfs/go-datastore) APIs?
+	- You can wrap those in `storage/dsadapter` and keep using them.
+	  You can also plug that into `linking.LinkSystem` to get higher level IPLD operations.
+	- Or if you were only using datastore because of some specific implementation of it, like, say `flatfs`?
+	  Then check out the possibility of moving all the way directly to new implementations like `storage/fsstore`.
+- Were you using [ipfs/go-ipfs-blockstore](https://pkg.go.dev/github.com/ipfs/go-ipfs-blockstore) APIs?
+	- You can wrap those in `storage/bsadapter` and keep using them.
+	  You can also plug that into `linking.LinkSystem` to get higher level IPLD operations.
+	  (This is almost exactly the same; we've just simplified in the interface, made it easier to implement, and cleaned up inconsistencies with the other interfaces in this migration guide which were already very very similar.)
+- Were you using [ipfs/go-blockservice](https://pkg.go.dev/github.com/ipfs/go-blockservice) APIs?
+	- You can wrap those in `storage/bsrvadapter` and keep using them.
+	  You can also plug that into `linking.LinkSystem` to get higher level IPLD operations.
+		- Be judicious about whether you actually want to do this.
+		  Plugging in the potential to experience unknown network latencies into code that's expecting predictable local lookup speeds
+		  may have undesirable performance outcomes.  (But if you want to do it, go ahead...)
+- Were you using [ipfs/go-ipld-format.DAGService](https://pkg.go.dev/github.com/ipfs/go-ipld-format#DAGService)?
+	- If you're using the `Add` and `Get` methods: those are now on `linking.LinkSystem`, as `Store` and `Load`.
+	- If you're using the `*Many` methods: those don't have direct replacements;
+	  we don't use APIs that force that style of future-aggregation anymore, because it's bad for pipelining.
+	  Just use `Store` and `Load`.
+- Were you using [ipfs/go-ipld-format.Node](https://pkg.go.dev/github.com/ipfs/go-ipld-format#Node)?
+	- That's actually a semantic doozy -- we use the word "node" _much_ differently now.
```

Comment:
whoa boy, yeah, is it any wonder this continues to confuse people if we have multiple uses of the word that mean something entirely different but so closely?

Reply:
Yeah, I wish... this was different, somehow.  But most of those wishes seem to involve time machines at this point.  Alas.